### PR TITLE
Cy 2980 add combine wagons command

### PR DESCRIPTION
**Pull request by AdarShaked**

### CODE REVIEW

File: wagon.py [hunk 1]
```diff
@@ -1011,6 +1011,94 @@ def repair(source, validate_archive=False):
...
 
 
+def _combine_wheels(processed_source_a, processed_source_b, metadata_a,
+                    metadata_b):
+    """ The workdir of the new wagon is processed_source_a dir  """
+    wheels_path_a = os.path.join(processed_source_a, DEFAULT_WHEELS_PATH)
+    wheels_path_b = os.path.join(processed_source_b, DEFAULT_WHEELS_PATH)
+
+    for wheel in _get_downloaded_wheels(wheels_path_b):
+        if not os.path.isfile(os.path.join(wheels_path_a, wheel)):
```
Comment (EarthmanT) [inline wagon.py:1021]: Is there a case where the file exists?

File: wagon.py [hunk 2]
```diff
@@ -1011,6 +1011,94 @@ def repair(source, validate_archive=False):
...
+            metadata_a["wheels"].append(wheel)
+    logger.info("finished combining wheels!")
+    return metadata_a
+
+
+def _compare_metadatas_field(metadata_a, metadata_b, field):
+    if not (metadata_a[field] == metadata_b[field]):
+        logger.info(
+            "{0} is different between accepted wagons,"
+            " cant combine them.".format(field))
```
Comment (EarthmanT) [inline wagon.py:1034]: can't

File: wagon.py [hunk 3]
```diff
@@ -1011,6 +1011,94 @@ def repair(source, validate_archive=False):
...
+            logger.info("copying wheel {0}".format(wheel))
+            shutil.copy2(src=os.path.join(wheels_path_b, wheel),
+                         dst=wheels_path_a)
+            metadata_a["wheels"].append(wheel)
+    logger.info("finished combining wheels!")
+    return metadata_a
+
+
+def _compare_metadatas_field(metadata_a, metadata_b, field):
+    if not (metadata_a[field] == metadata_b[field]):
```
Comment (EarthmanT) [inline wagon.py:1031]: why not `if metadata_a[field] != metadata_b[field]:`

File: wagon.py [hunk 4]
```diff
@@ -1011,6 +1011,94 @@ def repair(source, validate_archive=False):
...
+            "{0} is different between accepted wagons,"
+            " cant combine them.".format(field))
+        return False
+    return True
+
+
+def _validate_combine(metadata_a, metadata_b):
+    fields_to_validate = ["build_server_os_properties",
+                          "created_by_wagon_version", "package_name",
+                          "package_version", "supported_platform"]
```
Comment (EarthmanT) [inline wagon.py:1042]: consider moving this list to a constant instead of declaring inside the function.

File: wagon.py [hunk 5]
```diff
@@ -1011,6 +1011,94 @@ def repair(source, validate_archive=False):
...
+    1. Extract the wagons and their metadata.
+    2. validate the wagons can be combined.
+    3. Add all wheels of the second wagon to the first one
+    (without duplications).
+    4. Update the metadata of the combined wagon with new wheels list and
+     supported python versions.
+    5. pack the  new wagon.
+    """
+    _assert_linux_distribution_exists()
+    logger.info('combining: {0} , {1}'.format(source_a, source_b))
```
Comment (EarthmanT) [inline wagon.py:1062]: `Combining`

File: wagon.py (hunk 1 above)
Comment (AdarShaked) [reply wagon.py:1021]: Sure. if the wheel is py2py3 and both of the wagons has is. Just save time and not copy it.

**Pull request closed without merging**